lib/OpenQA/Shared/Controller/Auth.pm

```diff
@@ -56,62 +54,82 @@ sub check {
     return undef;
 }
 
-sub auth {
-    my $self = shift;
-    my $log  = $self->app->log;
-    my $user;
-
-    my $reason = "Not authorized";
+sub auth ($self) {
```

Comment:
Maybe it is time to split the code into smaller functions.

Reply:
Fair point, i was a bit focused on keeping all the old logic that i ignored opportunities for making the code cleaner.

Reply:
It also makes sense to do this kind of refactoring as a 2nd step (e.g. 2nd commit or further PR).